src/data-workspace/validation/use-validation-result.js

```diff
@@ -0,0 +1,125 @@
+import { useCallback } from 'react'
+import { useIsMutating, useQueries } from 'react-query'
+import {
+    useDataSetId,
+    useOrgUnitId,
+    usePeriodId,
+} from '../../context-selection/index.js'
+import { useApiAttributeParams } from '../../shared/index.js'
+
+const getValidationMetaDataQueryKey = (datasetId) => {
```

Comment:
`getValidationMetaDataQueryKey` as well as `getValidationQueryKey` should go into `src/data-workspace/query-key-factory.js`.